src/hooks/useHandleIcon.ts

```diff
@@ -0,0 +1,50 @@
+import { RefObject, useEffect, useState, ReactNode, useCallback } from "react";
```

Comment:
can you please explain purpose of this hook either by replying here or directly commenting in code?

Reply:
as much as I understood it adds or removes the icon on hover, cant we reach same implementation by the mix of CSS hoverr and transition effect?

Reply:
The solution looks simple, but actually to change the icon color(Which was created like a component) i think we need a call for this